conpot/core/attack_session.py

```diff
@@ -26,7 +26,7 @@
 # one instance per connection
 
 class AttackSession(object):
-    def __init__(self, protocol, source_ip, source_port, destination_ip, destination_port, databus, log_queue):
```

Comment:
afaik databus is a really important component of the attack session. We need to keep track of what changes are made to the registers etc. Can you comment on why do you think we should remove it?

Reply:
It should be removed because it is not used in in `AttackSession` in any way. It may seem useful, but all work is actually delegated to the LogWorker (see `self.log_queue` in `add_event` further down).

You can see that after the small refactor there isn't much code left in this file. `AttackSession`s really are fairly short-lived (albeit important) objects that do not need to carry around too much baggage.

Reply:
The original motivation behind this change was that better decoupling leads to better testability.